hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/ZookeeperBasedRepository.java

```diff
@@ -89,21 +90,30 @@ protected boolean pathExists(String path) {
     private <T> Optional<T> readFrom(String path, ThrowingReader<T> supplier, boolean quiet) {
         try {
             byte[] data = zookeeper.getData().forPath(path);
-            return Optional.of(supplier.read(data));
+            if (ArrayUtils.isNotEmpty(data)) {
+                return Optional.of(supplier.read(data));
+            }
         } catch (JsonMappingException malformedException) {
             if (quiet) {
                 logger.warn("Unable to read data from path {}", path, malformedException);
+                return Optional.empty();
             } else {
                 throw new MalformedDataException(path, malformedException);
             }
         } catch (Exception exception) {
             if (quiet) {
                 logger.warn("Unable to read data from path {}", path, exception);
+                return Optional.empty();
             } else {
                 throw new InternalProcessingException(exception);
             }
         }
-        return Optional.empty();
+        if (quiet) {
+            logger.warn("No data at path {}", path);
```

Comment:
This log can live inside a `try` block and all other `return Optional.empty();` in `catch` blocks will be unneeded.

Reply:
Ok, simplified this method